src/storage/txn/process.rs

```diff
@@ -478,12 +481,37 @@ fn process_write_impl<S: Snapshot, L: LockManager>(
             mutations,
             primary,
             start_ts,
-            options,
+            mut options,
             ..
         } => {
-            let mut txn = MvccTxn::new(snapshot, start_ts, !cmd.ctx.get_not_fill_cache())?;
-            let mut locks = vec![];
+            let mut scan_mode = None;
             let rows = mutations.len();
+            if options.for_update_ts.is_zero() && rows > FORWARD_MIN_MUTATIONS_NUM {
+                // TiKV client must sort keys in ascending order.
+                let left_key = mutations.first().unwrap().key().clone().append_ts(start_ts);
+                let right_key = mutations
+                    .last()
+                    .unwrap()
+                    .key()
+                    .clone()
+                    .append_ts(TimeStamp::zero());
+                if !has_data_in_range(
+                    snapshot.clone(),
+                    CF_WRITE,
```

Comment:
how about CF_LOCK?

Reply:
If a MvccReader was created with `Forward`, it would use `seek]`' in column `CF_LOCK` instead of `get`, which means that it would skip nonexistent keys in CF_LOCK.

Reply:
Do you mean if it has sought a greater key, the following seeks will finish immediately because theses keys are likely to be less than the greater key?